code/game/turfs/simulated/floor/asteroid_floors.dm

```diff
@@ -58,6 +58,15 @@
 		if(1)
 			getDug()
 
+/turf/simulated/floor/plating/asteroid/proc/attempt_ore_pickup(obj/item/storage/bag/ore/S, mob/user)
+	if(!istype(S))
+		return
+
+	if(S.pickup_all_on_tile)
+		for(var/obj/item/stack/ore/O in contents)
+			O.attackby(S, user)
+			return
```

Comment:
This logic feels weird. Why does it return?

Reply:
It is weird. It's basically trying to find the first possible attempt to pickup items on the tile. When /obj/item/attackby gets called, it will iterate over all contents on the tile to attempt all possible item insertions:

https://github.com/ParadiseSS13/Paradise/blob/04e65c74969fb2c7f96db0a903747c18b990bfea/code/game/objects/items.dm#L413-L420

So it doesn't need to iterate over all contents here, it just needs to attempt it once. It's horrible and I didn't want to try and untangle it in this PR.

Reply:
Gotcha. Don't blame you for not wanting to touch that, just figured I'd ask.